workspaces/mi/mi-extension/src/ai-panel/auth.ts

```
@@ -16,130 +16,421 @@
+                case LoginMethod.ANTHROPIC_KEY:
+                    resolve(credentials.secrets.apiKey);
+                    return;
+            }
         } catch (error: any) {
-            const errMsg = "Error while signing in to MI AI! " + error?.message;
-            throw new Error(errMsg);
+            reject(error);
         }
-    }
-}
+    });
```

Biome (2.1.2)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)